prg_test.go

```diff
@@ -0,0 +1,100 @@
+package test
+
+import (
+	"math"
+	mrand "math/rand"
+	"testing"
+
+	. "github.com/bjartek/overflow"
+	"github.com/onflow/flow-go/crypto/random"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNextUInt64NewPRG(t *testing.T) {
+	o, err := OverflowTesting()
+	assert.NoError(t, err)
+
+	t.Run("Should generate uniform distribution", func(t *testing.T) {
+		// make sure n is a power of 2 so that there is no bias in the last class
+		// n is a random power of 2 (from 2 to 2^10)
+		n := 1 << (1 + mrand.Intn(10))
+		classWidth := (math.MaxUint64 / uint64(n)) + 1
+		var sampleSize int
+
+		// taken from BasicDistributionTest constants - hardcoding here is
+		/// fragile, but it's determined within flow-go/crypto/rand_utils
```

Comment:
I agree this part to get the `sampleSize` isn't ideal, I also missed that this should have been exported by the tools in `flow-go/crypto/random` just like `BasicDistributionTest`.